### PR TITLE
Fix  appveyor

### PR DESCRIPTION
**Pull request by Dignifiedquire**

### CODE REVIEW

**Comment by Dignifiedquire** [comment]
No idea why circle ci is unhappy :(

**Comment by Kubuxu** [comment]
It is happy for me :smile:

**Comment by whyrusleeping** [comment]
@chriscool could we get some advice here? the sharness tests fail on windows CI because of path separator issues (at least, thats what it seems here). What would be a good way to do this?

**Comment by chriscool** [comment]
As far as I can see the first error is this:

```
expecting success: 
	echo "Error: failed to take lock at $IPFS_PATH: permission denied" > init_fail_exp &&
	test_cmp init_fail_exp init_fail_out
 
> diff -u init_fail_exp init_fail_out
--- init_fail_exp	2015-12-29 21:34:41.168675500 +0000
+++ init_fail_out	2015-12-29 21:34:41.168675500 +0000
@@ -1 +1 @@
-Error: failed to take lock at /cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0020-init.sh/.badipfs: permission denied
+Error: mkdir /cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0020-init.sh/.badipfs: The system cannot find the path specified.
 
not ok 3 - ipfs init output looks good
```

As permissions on Windows are working differently than on Linux/MacOS. It could be expected that the error message would not be the same on Windows.

Maybe this could be tested on Windows and either the error message we test against could be changed when testing on Windows or if the test is not relevant it could be skipped on Windows (maybe with a UNIXPERM prerequisite or something).

I don't have access to a Windows machine here. I will try to have a look at other test that are failing though.

**Comment by whyrusleeping** [comment]
I think we should merge this in for now. It still doesnt pass appveyor tests, but its gets a little farther and we can move forward from there

**Comment by Dignifiedquire** [comment]
:+1: for merging at least actual tests are now failing

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: d2157f522303d046528e267a731cbf337d7c7bbd
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by: .* <.*>**

 * Commit: d586996d56603bca115e9acda033eebd3ba3f847
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by: .* <.*>**

 * Commit: 596fc036a8b8683a89aa93d74c5f4666f9a4ab03
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by: .* <.*>**

We ask for a few features in the commit message for Open Source licensing hygiene and commit message clarity.
`git commit --amend` can often help you quickly improve the commit message.
[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) and [a script](https://github.com/ipfs/community/blob/master/dev/tools/hooks/setup_commit_msg_hook.sh) are available to help in the long run.
Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23).

---
This message was auto-generated by https://gitcop.com

**Comment by Dignifiedquire** [comment]
@whyrusleeping just rebased and squashed. Can we get this merged now?

**Comment by jbenet** [comment]
@Dignifiedquire is it actually fixed? it says:

```
Cannot find the tests' local ipfs tool.
Please check test and ipfs tool installation.
Makefile:29: recipe for target 't0010-basic-commands.sh' failed
make[1]: *** [t0010-basic-commands.sh] Error 1
make[1]: Leaving directory '/cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness'
Makefile:59: recipe for target 'test_sharness_expensive' failed
make: *** [test_sharness_expensive] Error 2
Command exited with code 2
```
- https://ci.appveyor.com/project/ipfsbot/test/build/561

**Comment by Dignifiedquire** [comment]
It was :( but it looks like something went wrong/changed after the rebase.

**Comment by Dignifiedquire** [comment]
@jbenet that was close, luckily I had the fixed version on my other machine. Should be better now

File: test/dependencies/iptb/main.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
 )
 
+var setupOpt = func(cmd *exec.Cmd) {}
```
Comment (jbenet) [inline test/dependencies/iptb/main.go]: were these changes contributed to `iptb` proper?

Comment (Dignifiedquire) [inline test/dependencies/iptb/main.go]: Yes, this was just pulled in from `iptb`

**Comment by jbenet** [comment]
Ok great! this LGTM. down to merge it before fixing the bugs. But before-- the iptb stuff should be updated in the iptb repo (https://github.com/whyrusleeping/iptb) and then update the dep here.

File: test/dependencies/iptb/main.go (hunk 1 above)
Comment (jbenet) [inline test/dependencies/iptb/main.go]: ah ok!

Comment (Dignifiedquire) [inline test/dependencies/iptb/main.go]: https://github.com/whyrusleeping/iptb/commit/b21f95b87a2275911fda072ee916dd2da72b436a is the commit I think

**Comment by jbenet** [comment]
Thanks @Dignifiedquire, @whyrusleeping, @chriscool, @Kubuxu, and whoever else was involved -- it's a big deal to get this working :)

**Comment by Dignifiedquire** [comment]
You have mostly to thank 32C3 dark room hack sessions :D

**Pull request merged**